src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,179 @@
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                statement.setLong(1, userId);
+                ResultSet resultSet = statement.executeQuery();
+                while (resultSet.next()) {
+                    userOrders.add(getOrderFromResultSet(resultSet));
+                }
+            }
+            for (Order order : userOrders) {
+                order.setProducts(getProductListByOrderId(order.getId()));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't find orders by userId: " + userId, e);
+        }
+        return userOrders;
+    }
+
+    private List<Product> getProductListByOrderId(Long orderId) throws SQLException {
+        List<Product> orderProducts = new ArrayList<>();
+        String query = "SELECT product_id, product_name,price FROM products "
+                + "JOIN orders_products USING (product_id) "
+                + "WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, orderId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long id = resultSet.getLong("product_id");
+                String productName = resultSet.getString("product_name");
+                double price = resultSet.getDouble("price");
+                Product product = new Product(productName, price);
+                product.setId(id);
+                orderProducts.add(product);
+            }
+            return orderProducts;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't get products from DB, orderId: " + orderId, e);
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Long cartId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        Order order = new Order(userId);
+        order.setId(cartId);
+        return order;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create provided order: " + order, e);
```

Comment:
It's better to pass its id instead of the whole order info because the product list might be too massive.